session.go

```diff
@@ -5212,3 +5226,67 @@ func hasErrMsg(d []byte) bool {
 	}
 	return false
 }
+
+// getRFC2253NameString converts from an ASN.1 structured representation of the certificate
+// to a UTF-8 string representation(RDN) and returns it.
+func getRFC2253NameString(certificate *x509.Certificate) string {
+	var RDNElementsASN1 = pkix.RDNSequence{}
+	asn1.Unmarshal(certificate.RawSubject, &RDNElementsASN1)
+
+	var RDNElementsString = []string{}
+	for i := len(RDNElementsASN1) - 1; i >= 0; i-- {
+		var nameAndValueList = []string{}
```

Comment:
This is not a critical path I think, but as a tip:

var nameAndValueList = make([]string,len(RDNElementsASN1[i))